test/snapshot/widget_snapshot_checker.cc

```diff
@@ -0,0 +1,185 @@
+/* Copyright 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/test/snapshot/widget_snapshot_checker.h"
+
+#include <string>
+
+#include "base/files/file_util.h"
+#include "base/memory/ref_counted_memory.h"
+#include "base/path_service.h"
+#include "base/threading/thread_restrictions.h"
+#include "brave/common/brave_paths.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "ui/compositor/test/draw_waiter_for_test.h"
+#include "ui/gfx/image/image.h"
+#include "ui/snapshot/snapshot.h"
+#include "ui/views/widget/widget.h"
+
+#if defined(USE_AURA)
+#include "ui/snapshot/snapshot_aura.h"
+#endif
+
+namespace {
```

Comment:
Maybe consider breaking out into a utility so that future snapshot tests can reuse the helpers